packages/angular/src/generators/directive/lib/normalize-options.ts

```diff
@@ -0,0 +1,54 @@
+import type { ProjectConfiguration, Tree } from '@nrwl/devkit';
+import {
+  joinPathFragments,
+  names,
+  readNxJson,
+  readProjectConfiguration,
+} from '@nrwl/devkit';
+import { parseName } from '../../utils/names';
+import type { Schema } from '../schema';
+
+export function normalizeOptions(tree: Tree, options: Schema) {
+  const { prefix, projectType, root, sourceRoot } = readProjectConfiguration(
+    tree,
+    options.project
+  ) as ProjectConfiguration & { prefix?: string };
+
+  const projectSourceRoot = sourceRoot ?? joinPathFragments(root, 'src');
+  const { name, path: namePath } = parseName(options.name);
+
+  const path =
+    options.path ??
+    joinPathFragments(
+      projectSourceRoot,
+      projectType === 'application' ? 'app' : 'lib',
+      namePath
+    );
+
+  const selector =
+    options.selector ?? buildSelector(tree, name, options.prefix, prefix);
```

Comment:
Could this be extracted to a shared util for other generators? 

This same logic would be used for pipe + component too right?

We'd need to change buildSelector to potentially accept a final argument, something like

```ts
export function buildSelector(Tree, string, string, string, keyof ReturnType<names>)

// then it could be used like

buildSelector(tree, name, options.prefix, projectPrefix, 'propertyName')
```

Reply:
Yeah. As mentioned in the description, I'll make the logic consistent in follow-up PRs. That includes extracting and making reusable whatever makes sense.